pkgs/tools/package-management/nix/default.nix

```diff
@@ -242,6 +253,19 @@ in lib.makeExtensible (self: ({
     hash = "sha256-EtL6M0H5+0mFbFh+teVjm+0B+xmHoKwtBvigS5NMWoo=";
   };
 
+  nix_2_20 = common {
+    version = "2.20.2";
+    hash = "sha256-avydDtF3k+FfBMh60TIVqoqPy0mjqwjZnw1gFba+7T0=";
+    patches = [
```

Comment:
What about the read-only regression fix (https://github.com/NixOS/nix/issues/10040)?

Reply:
I believe we're waiting fo the next tag.

Reply:
we can fetch the patch and when I am not doing a typo, the tests should succeed.